src/BloomExe/Browser.cs

```diff
@@ -350,10 +357,16 @@ protected override void OnLoad(EventArgs e)
 			GeckoPreferences.User["layout.spellcheckDefault"] = 0;
 
 			RaiseGeckoReady();
-	   }
+
+			Debug.Assert(ParentForm != null);
+			ParentForm.Activated += (sender, ee) => _updateCommandsTimer.Enabled = true;
+			ParentForm.Deactivate += (sender, ee) => _updateCommandsTimer.Enabled = false;
```

Comment:
Good point. I did it this way because that was the easiest I got it to work, but you're right that this isn't ideal. Should be enough to enable/disable on the focus events.

However, when I tried it it turns out that geckofx seems to mess with the focus: Browser.OnGotFocus never gets called (IIRC), so I overrode Browser.WndProc and enabled the timer when we get the WM_SETFOCUS message. However, immediately afterwards OnLostFocus gets called, so something is fishy there. Will do more debugging next week.